### PR TITLE
add zocl dma operation programming guide doc

### PR DESCRIPTION
**Pull request by xdavidz**
Please review it here:

file:///proj/xsjhdstaff2/davidzha/XRT_zocl/build/Release/runtime_src/doc/html/zocl_design.html

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_dma.h [hunk 1]
```diff
@@ -22,11 +22,34 @@
...
+ * @dma_flags: indicate if DMA returns error
+ * @dma_chan: DMA Channel, acquired before use DMA
+ * @dma_cookie: DMA Engine cookie
+ * @dma_done: DMA completion opaque
+ * @dma_func: call back function when dma complete
+ * @dma_arg: private data for dma_func call back function
+ *
+ * Caller can set dma_func and dma_arg for handling subsequent asynchronous
+ * operations, when DMA is completed dma_func will be called with dma_arg. DMA
+ * errors is being returned via zocl_dma_complet_cb argument ret.
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_dma.h]: zocl_dma_complete_cb

Comment (xdavidz) [reply src/runtime_src/core/edge/drm/zocl/zocl_dma.h]: thanks, fixed.

**Comment by xdavidz** [comment]
retest please

**Comment by maxzhen** [comment]
close for now until the conflict is resolved.

**Pull request closed without merging**